src/backend/distributed/planner/multi_explain.c

```diff
@@ -99,10 +173,655 @@ MultiExplainOneQuery(Query *query, IntoClause *into, ExplainState *es,
 		appendStringInfo(es->str, "%s\n", formattedPlanString);
 	}
 
-	/* if explain printing isn't enabled, print error only after planning */
-	if (!ExplainMultiLogicalPlan && !ExplainMultiPhysicalPlan)
+	if (!ExplainDistributedQueries)
 	{
 		appendStringInfo(es->str, "explain statements for distributed queries ");
+		appendStringInfo(es->str, "are not enabled\n");
+		return;
+	}
+
+	workerJob = multiPlan->workerJob;
+
+	if (list_length(workerJob->dependedJobList) > 0)
+	{
+		appendStringInfo(es->str, "explain statements for complex queries ");
 		appendStringInfo(es->str, "are currently unsupported\n");
```

Comment:
We chose to accept this restriction because of the amount of effort, and the comparatively low demand for complex task tracker stuff?